pip/qsharp/_qsharp.py

```diff
@@ -37,16 +37,54 @@ def _repr_mimebundle_(
         return {"application/x.qsharp-config": self._config}
 
 
-def init(target_profile: TargetProfile = TargetProfile.Unrestricted) -> Config:
+def init(
+    *, target_profile: TargetProfile = TargetProfile.Unrestricted, project_root=None
```

Comment:
Project root is missing type and has an implict optional type. Specity Optional[T] for what path join needs